tests/unit/EntityId/PrefixingEntityIdParserTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Tests\EntityId;
+
+use Wikibase\DataModel\Entity\BasicEntityIdParser;
+use Wikibase\DataModel\Entity\EntityIdParsingException;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Services\EntityId\PrefixingEntityIdParser;
+
+/**
+ * @covers Wikibase\DataModel\Services\EntityId\SuffixEntityIdParser
+ *
+ * @license GPL-2.0+
+ */
+class PrefixingEntityIdParserTest extends \PHPUnit_Framework_TestCase {
+
+	/**
+	 * @dataProvider validInputProvider
+	 */
+	public function testParse( $prefix, $input, $expected ) {
+		$parser = new PrefixingEntityIdParser( $prefix, new BasicEntityIdParser() );
+		$this->assertEquals( $expected, $parser->parse( $input ) );
+	}
+
+	public function validInputProvider() {
+		return array(
+			array( 'wikidata:', 'Q1337', new ItemId( 'wikidata:Q1337' ) ),
+			array( 'foo:', 'P123', new PropertyId( 'foo:P123' ) ),
+			array( 'foo:', 'bar:Q1337', new ItemId( 'foo:bar:Q1337' ) ),
```

Comment:
I assume we are talking about the same.

This code allows prefixed EntityIDs with **multiple** prefixes. As far as I remember we never talked about this. I strongly suggest to not allow this anywhere in any code as long as we don't really need this, and have a specification for this that covers questions like the ones I raised above.